utils/dcanalyse

```diff
@@ -0,0 +1,265 @@
+import pandas as pd
+import yaml
+
+NUM_ACCEL = 8
+
+
+def G2_M_S_2(g: float) -> float:
+    return g * 9.80665
+
+
+@dataclass
+class Accelerometer:
+    bias: float
+    offset: float
+    sensitivity: float
+    scalar: float
+
+    def calibrate(self, raw_value: float) -> float:
+        return (
+            G2_M_S_2(((raw_value - self.bias) / self.sensitivity) * self.scalar)
+            + self.offset
+        )
+
+
+class Accelerometers:
+    def __init__(self, config: dict):
+        self.accelerometers: list[Accelerometer] = []
+
+        for acc in range(1, NUM_ACCEL + 1):
+            acc_config = config[f"Accelerometer{acc}"]
+            self.accelerometers.append(
+                Accelerometer(
+                    acc_config["Bias"],
+                    acc_config["Offset"],
+                    acc_config["Sensitivity"],
+                    acc_config["Scalar"],
+                )
+            )
+
+    def calibrate(self, raw_values: list[float]) -> list[float]:
+        ret: list[float] = []
+        for acc_index, acc in enumerate(self.accelerometers):
+            ret.append(acc.calibrate(raw_values[acc_index]))
+        return ret
+
+
+class Processor:
+    def __init__(self, width: int, depth: int, stats: list[int], config_filename: str):
+        self.width = width
+        self.depth = depth
+        self.stats = stats
+
+        self.counts = [0] * width
+        self.lower = None
+        self.upper = None
+
+        self.stats_raw: dict[int, list[list[float]]] = dict([(k, []) for k in stats])
+
+        for ax in "ABCDEFGH12345678xyz":
+            setattr(self, "stats_" + ax, dict([(k, []) for k in stats]))
+
+        with open(config_filename) as config_file:
+            config = yaml.safe_load(config_file)["AccelerometerSettings"]
+            self.x_distance = config["AngularAccelerationXDistance"]
+            self.y_distance = config["AngularAccelerationYDistance"]
+            self.z_distance = config["AngularAccelerationZDistance"]
+
+            self.accelerometers = Accelerometers(config)
+
+        print(str(self.accelerometers.accelerometers[0]))
+
+        logging.info("Loaded config from %s", config_filename)
+
+    def get_xyz(self, acc_data: list[float]) -> tuple[float, float, float]:
+        return (
+            float(np.degrees((acc_data[5] - acc_data[7]) / self.x_distance)),
+            float(np.degrees((acc_data[2] - acc_data[0]) / self.y_distance)),
+            float(np.degrees((acc_data[4] - acc_data[0]) / self.z_distance)),
+        )
+
+    def process(self, line: int, raw_data: list[float]):
+        calibrated = self.accelerometers.calibrate(raw_data)
+        xyz = self.get_xyz(calibrated)
+        for stat_size in self.stats:
+            self.stats_raw[stat_size].append(raw_data + calibrated + list(xyz))
+            if len(self.stats_raw[stat_size]) >= stat_size:
+                data = pd.DataFrame(
+                    self.stats_raw[stat_size],
+                    columns=(
```

Comment:
I think ``columns = tuple("ABCDEFGH12345678xyz")`` would work here as well, and be more compact/readable.  Also, that string is used multiple places, so maybe a constant to avoid typos might be good.